### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by fbeetge** [CoreTweet/CoreTweet#189] [Q&A] [answered]
Hi
I have an app using Auth 1.0 that automatically posts to Twitter when records are inserted into a database.  I would like to port it to V1.1, or even better, V2.   Currently using C#/VB.net.

I have followed the supplied examples and posting to my account is extremely easy, except for 2 issues specific to my application:
1) The example requires a human to authorize the session by clicking a button, and 
2) Getting the PIN code from the reply URL (completely outside of my app) into the app before being allowed to post the Tweet.

Is there a way that one can use this library, but do the authentication entirely within my application (like with Auth V1.0)?

Kind Regards
Francois J Beetge

**Comment by cannorin** [comment, marked as answer]
Hi, in general you have to use a [callback URL](https://developer.twitter.com/en/docs/apps/callback-urls) for this. But the way to accomplish it really depends on how you are running your app.

If you are running it on some web server, you can just use it to receive a callback. If you are running it on desktop, you could create a temporary webserver on localhost and let Twitter callback to `127.0.0.1`, or use a app-specific protocol (`your-app://`) and let the browser pass it to your app.

This may be the optimal way to authenticate for end users, but if you are the only person using your app then it would only add extra complexity to your app and it would be better to keep using PINs.
